runtime/src/iree/hal/drivers/vulkan/vulkan_device.cc

```diff
@@ -36,6 +36,96 @@
 
 using namespace iree::hal::vulkan;
 
+//===----------------------------------------------------------------------===//
+// RenderDoc integration
+//===----------------------------------------------------------------------===//
+
+// Configure cmake with -DIREE_ENABLE_RENDERDOC_PROFILING=ON in order to
+// enable profiling support. This should be left off in production builds to
+// avoid introducing a backdoor.
+#if defined(IREE_HAL_VULKAN_HAVE_RENDERDOC)
+
+// NOTE: C API, see https://renderdoc.org/docs/in_application_api.html.
+// When compiled in the API will no-op itself if not running under a RenderDoc
+// capture context (renderdoc.dll/so already loaded).
+#include "third_party/renderdoc/renderdoc_app.h"
+
+typedef RENDERDOC_API_1_5_0 RENDERDOC_API_LATEST;
+
+// Returns a handle to the RenderDoc API when it is hooking the process.
+// Returns NULL when RenderDoc is not present (or valid).
+static RENDERDOC_API_LATEST* iree_hal_vulkan_query_renderdoc_api(
+    VkInstance instance) {
+  pRENDERDOC_GetAPI RENDERDOC_GetAPI = NULL;
+#if defined(IREE_PLATFORM_WINDOWS)
+
+  // NOTE: RenderDoc only supports hooking so we can't use LoadLibrary - if
+  // we're going to use RenderDoc its library must already be loaded.
+  if (HMODULE hook_module = GetModuleHandleA("renderdoc.dll")) {
```

Comment:
Is this DLL easy to discover this way? See https://github.com/iree-org/iree/pull/2131

Reply:
It's not needed here as we aren't loading the library and instead just fetching the handle if already loaded. GetModuleHandleA scans a global flattened module list where paths don't mater. Loading the library after we've touched any Vulkan files won't work and we don't want to be doing this unconditionally (why it's unsupported by RenderDoc - same reason the posix path uses RTLD_NOLOAD). `renderdoccmd capture iree-run-module ...` can be used to capture from command lines (I'll add that to docs).

Reply:
Added docs on `renderdoccmd capture`.